lib/cli/driver.js

```diff
@@ -262,9 +273,88 @@ class DriverCommand {
     return this.config.installedDrivers;
   }
 
-  update (/*{driver}*/) {
-    // TODO implementation
-    return {};
+  async update ({driver, force}) {
+    const all = driver === 'all';
+    // if we're specifically requesting an update for a driver, make sure it's installed
+    if (!all && !this.config.isInstalled(driver)) {
+      throw new Error(`Driver '${driver}' was not installed, so can't be updated`);
+    }
+    const driversToUpdate = all ? Object.keys(this.config.installedDrivers) : [driver];
+    const errors = {};
+    const updates = {};
+    for (const d of driversToUpdate) {
+      try {
+        await spinWith(this.isJsonOutput, `Checking if driver '${d}' is updatable`, () => {
+          if (this.config.installedDrivers[d].installType !== INSTALL_TYPE_NPM) {
+            throw new Error(`d '${d}' was not installed via npm, so we can't check for updates`);
+          }
+        });
+        const update = await spinWith(this.isJsonOutput, `Checking if driver '${d}' needs an update`, async () => {
+          const update = await this.checkForDriverUpdate(d);
+          if (!(update.safeUpdate || update.unsafeUpdate)) {
+            throw new Error(`Driver '${d}' had no updates available`);
+          }
+          return update;
+        });
+        if (!force && !update.safeUpdate) {
+          throw new Error(`Driver '${d}' has a major revision update ` +
+                          `(${update.current} => ${update.unsafeUpdate}), which could include ` +
+                          `breaking changes. If you want to apply this update, re-run with --force`);
+        }
+        const updateVer = force && update.unsafeUpdate ? update.unsafeUpdate : update.safeUpdate;
+        await spinWith(this.isJsonOutput, `Updating driver '${d}' from ${update.current} to ${updateVer}`, async () => {
+          await this.updateDriver(d, updateVer);
+        });
+        updates[d] = {from: update.current, to: updateVer};
+      } catch (err) {
+        errors[d] = err;
+      }
+    }
+    if (this.isJsonOutput) {
+      return updates;
+    }
+    log(this.isJsonOutput, 'Update report:');
+    for (const [d, update] of _.toPairs(updates)) {
+      log(this.isJsonOutput, `- Driver ${d} updated: ${update.from} => ${update.to}`.green);
+    }
+    for (const [d, err] of _.toPairs(errors)) {
+      if (/no updates available/.test(err)) {
+        log(this.isJsonOutput, `- Driver ${d} had no updates available`.yellow);
+      } else {
+        log(this.isJsonOutput, `- Driver ${d} failed to update: ${err}`.red);
+      }
+    }
+    return updates;
+  }
+
+  async checkForDriverUpdate (driver) {
+    // TODO decide how we want to handle beta versions?
```

Comment:
beta should be counted in the same manner as major updates - we should only allow it if it is explicitly enabled from the command line or in the config

Reply:
it also makes sense to assign it per-driver

Reply:
here is the question i had in mind:

Assume the user has version 1.2.3 of a driver installed. Now, there is also version 2.0.0 out, as well as 2.0.1-beta.

Which one would we expect Appium to update to?